akka-stream-tests/src/test/scala/akka/stream/scaladsl/ActorRefSinkSpec.scala

```diff
@@ -33,16 +34,14 @@ class ActorRefSinkSpec extends AkkaSpec {
     }
 
     "cancel stream when actor terminates" in {
-      val publisher = StreamTestKit.PublisherProbe[Int]()
       val fw = system.actorOf(Props(classOf[Fw], testActor).withDispatcher("akka.test.stream-dispatcher"))
-      Source(publisher).runWith(Sink.actorRef(fw, onCompleteMessage = "done"))
-      val autoPublisher = new StreamTestKit.AutoPublisher(publisher)
-      autoPublisher.sendNext(1)
-      autoPublisher.sendNext(2)
+      val sourceRef = watch(Source.actorRef(10, OverflowStrategy.fail).to(Sink.actorRef(fw, onCompleteMessage = "done")).run())
+      sourceRef ! 1
+      sourceRef ! 2
       expectMsg(1)
       expectMsg(2)
       system.stop(fw)
-      autoPublisher.subscription.expectCancellation()
+      expectTerminated(sourceRef)
```

Comment:
this does not test the same thing any more, it does not verify "cancel stream when actor terminates"